File: single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        nums.sort()        
        for i in range(1, len(nums), 3):
            if nums[i-1] != nums[i]:
                return nums[i-1]
        
        if len(nums) % 3 != 0:
            return nums[-1]

File: single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int answer = 0;

        for (int bit = 0; bit < 32; bit++) {
            int count = 0;
            for (int num : nums) {
                if (num & (1 << bit)) {
                    count++;
                }
            }

            if (count % 3 != 0) {
                answer |= (1 << bit);
            }
        }

        return answer;
    }
};